src/uiTest/scala/com/virtuslab/gitmachete/uitest/UITestSuite.scala

```diff
@@ -80,42 +85,49 @@ class UITestSuite extends BaseGitRepositoryBackedIntegrationTestSuite(SETUP_WITH
-      managedBranches.toSeq.sorted)
-    intelliJ.project.toggleListingCommits()
-    var branchAndCommitRowsCount = intelliJ.project.refreshModelAndGetRowCount()
-    // 6 branch rows + 7 commit rows
-    Assert.assertEquals(13, branchAndCommitRowsCount)
-
-    // Let's slide out a root branch now
-    intelliJ.project.slideOutBranch("develop")
-    intelliJ.project.acceptBranchDeletionOnSlideOut()
-    branchAndCommitRowsCount = intelliJ.project.refreshModelAndGetRowCount()
-    // 5 branch rows (`develop` is no longer there) + 3 commit rows
-    // (1 commit of `allow-ownership-link` and 3 commits of `call-ws` are all gone)
-    Assert.assertEquals(8, branchAndCommitRowsCount)
-
-    intelliJ.project.checkoutBranch("develop")
-    intelliJ.project.slideOutBranch("call-ws")
-    intelliJ.project.rejectBranchDeletionOnSlideOut()
-    branchAndCommitRowsCount = intelliJ.project.refreshModelAndGetRowCount()
-    // 4 branch rows (`call-ws` is also no longer there) + 3 commit rows
-    Assert.assertEquals(7, branchAndCommitRowsCount)
+    //TODO (#830): try ... catch block to discover why the SocketTimeoutException occurs
+    try {
+      intelliJ.project.openGitMacheteTab()
+      overwriteMacheteFile(
+        """develop
+          |  non-existent
+          |    allow-ownership-link
+          |      build-chain
+          |    non-existent-leaf
+          |  call-ws
+          |non-existent-root
+          |master
+          |  hotfix/add-trigger""".stripMargin
+      )
+      val managedBranches = intelliJ.project.refreshModelAndGetManagedBranches()
+      // Non-existent branches should be skipped while causing no error (only a low-severity notification).
+      Assert.assertEquals(
+        Seq("allow-ownership-link", "build-chain", "call-ws", "develop", "hotfix/add-trigger", "master"),
+        managedBranches.toSeq.sorted)
+      intelliJ.project.toggleListingCommits()
+      var branchAndCommitRowsCount = intelliJ.project.refreshModelAndGetRowCount()
+      // 6 branch rows + 7 commit rows
+      Assert.assertEquals(13, branchAndCommitRowsCount)
+
+      // Let's slide out a root branch now
+      intelliJ.project.slideOutBranch("develop")
+      intelliJ.project.acceptBranchDeletionOnSlideOut()
+      branchAndCommitRowsCount = intelliJ.project.refreshModelAndGetRowCount()
+      // 5 branch rows (`develop` is no longer there) + 3 commit rows
+      // (1 commit of `allow-ownership-link` and 3 commits of `call-ws` are all gone)
+      Assert.assertEquals(8, branchAndCommitRowsCount)
+
+      intelliJ.project.checkoutBranch("develop")
+      intelliJ.project.slideOutBranch("call-ws")
+      intelliJ.project.rejectBranchDeletionOnSlideOut()
+      branchAndCommitRowsCount = intelliJ.project.refreshModelAndGetRowCount()
+      // 4 branch rows (`call-ws` is also no longer there) + 3 commit rows
+      Assert.assertEquals(7, branchAndCommitRowsCount)
+    } catch {
+      case e: Exception =>
+        saveToFileThreadDump()
```

Comment:
Nit:
```suggestion
        saveThreadDumpToFile()
```

`{give,save,tell,...} X to Y`, not `<verb> to Y X`